src/software/simulation/firmware_object_deleter.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
```

Comment:
Might as well move all existing deleters here while we're at it. At the very least there's a `PrimitiveManagerDeleter` in `simulator_robot.h` that can be moved, and `FirmwareBallDeleter`